src/objective-c/tests/CoreCronetEnd2EndTests/CoreCronetEnd2EndTests.m

```diff
@@ -32,26 +32,349 @@
+                                                           grpc_end2end_test_fixture *f, grpc_channel_args *server_args) {
+  grpc_ssl_pem_key_cert_pair pem_cert_key_pair = {test_server1_key,
+    test_server1_cert};
+  grpc_server_credentials *ssl_creds =
+  grpc_ssl_server_credentials_create(NULL, &pem_cert_key_pair, 1, 0, NULL);
+  if (fail_server_auth_check(server_args)) {
+    grpc_auth_metadata_processor processor = {process_auth_failure, NULL, NULL};
+    grpc_server_credentials_set_auth_metadata_processor(ssl_creds, processor);
+  }
+  chttp2_init_server_secure_fullstack(f, server_args, ssl_creds);
+}
+
+/* All test configurations */
+
+static grpc_end2end_test_config configs[] = {
+  {"chttp2/simple_ssl_fullstack",
+    FEATURE_MASK_SUPPORTS_DELAYED_CONNECTION |
+    FEATURE_MASK_SUPPORTS_PER_CALL_CREDENTIALS,
+    chttp2_create_fixture_secure_fullstack,
+    cronet_init_client_simple_ssl_secure_fullstack,
+    chttp2_init_server_simple_ssl_secure_fullstack,
+    chttp2_tear_down_secure_fullstack},
+};
+
+
+
 
 @interface CoreCronetEnd2EndTests : XCTestCase
 
 @end
 
 @implementation CoreCronetEnd2EndTests
 
-- (void)testCoreCronetEnd2End {
-  char *argv[] = {"h2_ssl"};
-
-  // This main() function is not the entry point of this test case; it
-  // refers to that in h2_ssl_cronet.m. We can use it because XCode
-  // builder does not use main() as the entry point for a test.
-  // Since h2_ssl_cronet.m is derived from h2_ssl.c in the core end2end
-  // test fixture, we preserves the fixture structure (in particular
-  // the main() function here) so that another fixture can easily
-  // replace the fixture h2_ssl_cronet, in case we need more tests in
-  // the future
-  main(1, argv);
+char *roots_filename;
+
+// The setUp() function is run before the test cases run and only run once
++ (void)setUp {
+  [super setUp];
+
+  FILE *roots_file;
+  size_t roots_size = strlen(test_root_cert);
+  
+  char *argv[] = {"CoreCronetEnd2EndTests"};
+  grpc_test_init(1, argv);
+  grpc_end2end_tests_pre_init();
+  
+  /* Set the SSL roots env var. */
+  roots_file = gpr_tmpfile("chttp2_simple_ssl_fullstack_test", &roots_filename);
+  GPR_ASSERT(roots_filename != NULL);
+  GPR_ASSERT(roots_file != NULL);
+  GPR_ASSERT(fwrite(test_root_cert, 1, roots_size, roots_file) == roots_size);
+  fclose(roots_file);
+  gpr_setenv(GRPC_DEFAULT_SSL_ROOTS_FILE_PATH_ENV_VAR, roots_filename);
+  
+  grpc_init();
+  
+}
+
+// The tearDown() function is run after all test cases finish running
++ (void)tearDown {
+  grpc_shutdown();
+  
+  /* Cleanup. */
+  remove(roots_filename);
+  gpr_free(roots_filename);
+  
+  [super tearDown];
+}
+
+- (void)testIndividualCase:(char*)test_case {
+  char *argv[] = {"h2_ssl", test_case};
+  
+  for (int i = 0; i < sizeof(configs) / sizeof(*configs); i++) {
+    grpc_end2end_tests(sizeof(argv) / sizeof(argv[0]), argv, configs[i]);
+  }
+}
+
+- (void)testBadHostname {
+  [self testIndividualCase:"bad_hostname"];
```

Comment:
To help your future sanity, add a TODO to do instead:

```objective-c
  NSString *testName = [[NSStringFromSelector(_cmd) substringFromIndex:4] upperCamelAsLowerUnderscore];
  [self testIndividualCase:testName];
```
Then you can just copy the method body verbatim to all the test methods, without having to do the conversion yourself.

It'd need, at the beginning of the file:
```objective-c
@interface NSString (CaseConversion)
- (NSString *)upperCamelAsLowerUnderscore;
@end

@implementation NSString (CaseConversion)
- (NSString *)upperCamelAsLowerUnderscore {
  // This is why it's a TODO :D
}
@end
```